server/sonar-web/src/main/js/apps/settings/components/inputs/SimpleInput.js

```diff
@@ -26,12 +26,29 @@ export default class SimpleInput extends React.PureComponent {
     ...defaultInputPropTypes,
     value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
     type: PropTypes.string.isRequired,
-    className: PropTypes.string.isRequired
+    className: PropTypes.string.isRequired,
+    onCancel: PropTypes.func,
+    onSave: PropTypes.func
   };
 
-  handleInputChange(e) {
-    this.props.onChange(e.target.value);
-  }
+  handleInputChange = event => {
+    this.props.onChange(event.currentTarget.value);
+  };
+
+  handleKeyDown = event => {
+    switch (event.keyCode) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this "switch" statement with "if" statements to increase readability. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS1301)